x-pack/packages/ml/response_stream/server/stream_factory.ts

```diff
@@ -132,6 +134,23 @@ export function streamFactory<T = unknown>(
     // otherwise check the integrity of the data to be pushed.
     if (streamType === undefined) {
       streamType = typeof d === 'string' ? 'string' : 'ndjson';
+
+      // This is a fix for ndjson streaming with proxy configurations
+      // that buffer responses up to 4KB in size. We keep track of the
+      // size of the response sent so far and if it's still smaller than
+      // FLUSH_PAYLOAD_SIZE then we'll push an additional keep-alive object
+      // that contains the flush fix payload.
+      if (flushFix && streamType === 'ndjson') {
+        setTimeout(function repeat() {
```

Comment:
Not sure if we need the outer setTimeout(fn, 0) Can be this achieved with just calling repeat()?

Reply:
You're right of course, updated in [5cc4de0](https://github.com/elastic/kibana/pull/165069/commits/5cc4de077e863009216b3d4b8c702425e835ef08)!